### PR TITLE
Fix java setup + minor improvements

### PR DESCRIPTION
**Pull request by SylvainJuge**
- remove maven build files + ignore them
- allow extra .env config + fix regressions
- use 2.0.0 upstream otel agent

### CODE REVIEW

File: Elastiflix/.gitignore [hunk 1]
```diff
@@ -8,3 +8,10 @@ node_modules/
 **/.idea
 dotnet*/obj
 .idea
+
+# maven target folder
```
Comment (SylvainJuge) [inline Elastiflix/.gitignore:12]: [for reviewer] those files are produced by the build and should be ignored.

File: Elastiflix/docker-compose-elastic-otel.yml [hunk 2]
```diff
@@ -96,6 +96,7 @@ services:
     ports:
       - "5003:5000"
     environment:
+      - ELASTIC_APM_SECRET_TOKEN=${ELASTIC_APM_SECRET_TOKEN}
```
Comment (SylvainJuge) [inline Elastiflix/docker-compose-elastic-otel.yml:99]: [for reviewer] for now the "manual" variant relies on this environment variable to explicitly deal with authentication, this is something that could be improved later to only rely on standard otel environment variables.

File: Elastiflix/docker-compose-elastic-otel.yml [hunk 3]
```diff
@@ -124,12 +125,13 @@ services:
...
       - OTEL_LOGS_EXPORTER=otlp
       - OTEL_TRACES_EXPORTER=otlp
+      - OTEL_JAVAAGENT_DEBUG=${OTEL_JAVAAGENT_DEBUG:-false}
       - REDIS_HOST=redis
       - TOGGLE_SERVICE_DELAY=${TOGGLE_SERVICE_DELAY:-0}
       - TOGGLE_CANARY_DELAY=${TOGGLE_CANARY_DELAY:-0}
       - TOGGLE_CANARY_FAILURE=${TOGGLE_CANARY_FAILURE:-0}
       # agent distribution, either 'otel' or 'elastic'
-      - AGENT_DISTRIBUTION=otel
+      - AGENT_DISTRIBUTION=${AGENT_DISTRIBUTION:-otel}
```
Comment (SylvainJuge) [inline Elastiflix/docker-compose-elastic-otel.yml:134]: [for reviewer] this allows to easily override those environment variables from the `.env`

File: Elastiflix/java-favorite-otel-auto/Dockerfile [hunk 4]
```diff
@@ -16,7 +16,7 @@ RUN mvn clean install
 USER root
 RUN apt-get update && apt-get install -y zip curl
 
-ARG OTEL_AGENT_VERSION=1.28.0
+ARG OTEL_AGENT_VERSION=2.0.0
```
Comment (SylvainJuge) [inline Elastiflix/java-favorite-otel-auto/Dockerfile:19]: [for reviewer] the elastic distro currently relies on this version, thus it allows to make the features and behavior consistent.

File: Elastiflix/java-favorite-otel-auto/start.sh [hunk 5]
```diff
@@ -14,5 +14,9 @@ case "${AGENT_DISTRIBUTION:-otel}" in
   ;;
 esac
 
-java -javaagent:${agent_path} \
--jar /usr/src/app/target/favorite-0.0.1-SNAPSHOT.jar --server.port=5000
```
Comment (SylvainJuge) [inline Elastiflix/java-favorite-otel-auto/start.sh:18]: [for reviewer] the extra `-` before the `-jar` argument was what made it not start properly.

**Review by david-luna — APPROVED** [review]

**Pull request merged**